src/main/java/sg/ncl/MainController.java

```diff
@@ -340,15 +345,20 @@ public String logout(HttpSession session) {
     
     @RequestMapping(value="/signup2", method=RequestMethod.GET)
     public String signup2(Model model) {
-    	// TODO get each model data and put into relevant ones
-    	model.addAttribute("loginForm", new LoginForm());
     	model.addAttribute("signUpMergedForm", new SignUpMergedForm());
     	return "signup2";
     }
     
     @RequestMapping(value="/signup2", method=RequestMethod.POST)
-    public String validateDetails(@ModelAttribute("loginForm") LoginForm loginForm, @ModelAttribute("signUpMergedForm") SignUpMergedForm signUpMergedForm) {
-    	// TODO get each model data and put into relevant ones
+    public String validateDetails(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "validateDetails" is 20 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)